datamodel/low/v3/server_test.go

```diff
@@ -55,6 +55,61 @@ variables:
 	for v := range n.Variables.Value.ValuesFromOldest() {
 		assert.NotNil(t, v.Value.GetKeyNode())
 		assert.NotNil(t, v.Value.GetRootNode())
+		assert.Nil(t, v.Value.GetExtensions())
+	}
+}
+
+func TestServerWithVariableExtension_Build(t *testing.T) {
+	yml := `url: https://pb33f.io
+description: high quality software for developers.
+variables:
+  var1: 
+    default: hello
+    description: a var
+    enum: [one, two]
+    x-transforms:
+        allowMissing: true`
+
+	var idxNode yaml.Node
+	_ = yaml.Unmarshal([]byte(yml), &idxNode)
+	idx := index.NewSpecIndex(&idxNode)
+
+	var n Server
+	err := low.BuildModel(idxNode.Content[0], &n)
+	assert.NoError(t, err)
+	assert.Nil(t, n.GetRootNode())
+
+	err = n.Build(context.Background(), nil, idxNode.Content[0], idx)
+	assert.NoError(t, err)
+	assert.NotNil(t, n.GetRootNode())
+	assert.Equal(t, "ec69dfcf68ad8988f3804e170ee6c4a7ad2e4ac51084796eea93168820827546",
+		low.GenerateHashString(&n))
+
+	assert.Equal(t, "https://pb33f.io", n.URL.Value)
+	assert.Equal(t, "high quality software for developers.", n.Description.Value)
+
+	variable := n.FindVariable("var1").Value
+	assert.Equal(t, "hello", variable.Default.Value)
+	assert.Equal(t, "a var", variable.Description.Value)
+
+	var_extensions := variable.Extensions.First()
+	assert.Equal(t, "x-transforms", var_extensions.Key().Value)
+	variable_pair := variable.Extensions.GetPair(var_extensions.Key())
+	assert.Equal(t, "allowMissing", variable_pair.Value.Value.Content[0].Value)
+	assert.Equal(t, "true", variable_pair.Value.Value.Content[1].Value)
```

Comment:
Thanks @daveshanley !
I did the best I could, but I think it looks ugly. However, it works.
If you have any tips for this part of the test, I'd be happy to study it and improve it.